here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/apis/ApiUtil.java

```diff
@@ -0,0 +1,213 @@
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.geojson.WebMercatorTile;
+import com.here.naksha.lib.core.models.geojson.implementation.namespaces.XyzNamespace;
+import com.here.naksha.lib.core.models.payload.events.QueryDelimiter;
+import com.here.naksha.lib.core.models.payload.events.QueryParameter;
+import com.here.naksha.lib.core.models.payload.events.QueryParameterList;
+import com.here.naksha.lib.core.models.storage.POp;
+import com.here.naksha.lib.core.models.storage.PRef;
+import com.here.naksha.lib.core.models.storage.ReadRequest;
+import com.here.naksha.lib.core.models.storage.SOp;
+import com.here.naksha.lib.core.util.ValueList;
+import com.here.naksha.lib.core.util.storage.RequestHelper;
+import com.vividsolutions.jts.geom.Geometry;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ApiUtil {
+
+  private static final int NONE = 0, OR = 1, AND = 2;
+
+  public static @NotNull SOp buildOperationForTile(final @NotNull String tileType, final @NotNull String tileId) {
+    try {
+      if (!tileType.equals("quadkey")) {
+        throw new XyzErrorException(XyzError.ILLEGAL_ARGUMENT, "Tile type " + tileType + " not supported");
+      }
+      final Geometry geo =
+          WebMercatorTile.forQuadkey(tileId).getAsPolygon().getGeometry();
+      return SOp.intersects(geo);
+    } catch (IllegalArgumentException ex) {
+      throw new XyzErrorException(XyzError.ILLEGAL_ARGUMENT, ex.getMessage());
+    }
+  }
+
+  public static @NotNull SOp buildOperationForBBox(
+      final double west, final double south, final double east, final double north) {
+    return SOp.intersects(RequestHelper.createBBoxEnvelope(west, south, east, north));
+  }
+
+  /**
+   * Function builds Property Operation (POp) based on "tags" supplied as API query parameter.
+   * We iterate through all the tag combination values provided in the query param.
+   *   For every tag combination concatenated with "," (COMMA) delimiter, we prepare OR list
+   *   For every tag combination concatenated with "+" (PLUS) delimiter, we prepare AND list
+   *   For any tag not part of any combination, we take it as part of OR condition
+   *
+   * So, for example if the input tag list is like this:
+   *   tags = one
+   *   tags = two,three
+   *   tags = four+five
+   *   tags = six,seven,eight+nine
+   *   tags = ten+eleven,twelve,thirteen
+   *   tags = fourteen
+   *   Then, we generate
+   * OR operation between:
+   *   - one
+   *   - (two OR three)
+   *   - (four AND five)
+   *   - (six OR seven)
+   *   - (eight AND nine)
+   *   - (ten AND eleven)
+   *   - (twelve OR thirteen)
+   *   - fourteen
+   *
+   * @param queryParams API query parameter from where "tags" needs to be extracted
+   * @return POp property operation that can be used as part of {@link ReadRequest}
+   */
```

Comment:
IMO this tag based POp query is non obvious - I think we should document it in swagger for API clients